### PR TITLE
FM-7782 - cleanup 

### PR DESCRIPTION
**Pull request by lionce**
- removed modules acceptance tests -> replaced by units
- removed service acceptance tests
- removed facts -> WIP
- fixed rubocop issues 
apache_cleanup

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -25,17 +25,6 @@
       it { is_expected.to be_installed }
     end
 
-    describe service($service_name) do
-      if fact('operatingsystem') == 'Debian' && fact('operatingsystemmajrelease') == '8'
-        pending 'Should be enabled - Bug 760616 on Debian 8'
-      elsif fact('operatingsystem') == 'SLES' && fact('operatingsystemmajrelease') == '15'
-        pending 'Should be enabled - MODULES-8379 `be_enabled` check does not currently work for apache2 on SLES 15'
```
Comment (tphoney) [inline spec/acceptance/class_spec.rb]: keep these service checks for class

File: spec/acceptance/itk_spec.rb [hunk 2]
```diff
@@ -49,15 +49,4 @@ class { 'apache':
...
   end
-
-  describe service(service_name) do
-    it { is_expected.to be_running }
-    if fact('operatingsystem') == 'Debian' && fact('operatingsystemmajrelease') == '8'
-      pending 'Should be enabled - Bug 760616 on Debian 8'
-    elsif fact('operatingsystem') == 'SLES' && fact('operatingsystemmajrelease') == '15'
-      pending 'Should be enabled - MODULES-8379 `be_enabled` check does not currently work for apache2 on SLES 15'
-    else
-      it { is_expected.to be_enabled }
```
Comment (tphoney) [inline spec/acceptance/itk_spec.rb]: put the service check back

File: spec/acceptance/vhost_spec.rb [hunk 3]
```diff
@@ -228,17 +217,6 @@ class { 'apache':
       apply_manifest(pp, catch_failures: true)
     end
 
-    describe service($service_name) do
-      if fact('operatingsystem') == 'Debian' && fact('operatingsystemmajrelease') == '8'
-        pending 'Should be enabled - Bug 760616 on Debian 8'
-      elsif fact('operatingsystem') == 'SLES' && fact('operatingsystemmajrelease') == '15'
-        pending 'Should be enabled - MODULES-8379 `be_enabled` check does not currently work for apache2 on SLES 15'
```
Comment (tphoney) [inline spec/acceptance/vhost_spec.rb]: put the service checks back in.

File: spec/acceptance/zzz_centos_scl_spec.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # "zzz_" got added to ensure, this test runs last, as it breaks the non-SCL httpd in later tests
```
Comment (tphoney) [inline spec/acceptance/zzz_centos_scl_spec.rb]: this becomes a unit test. delete this acceptance test !!!

File: spec/acceptance/mod_passenger_spec.rb [hunk 5]
```diff
@@ -1,14 +1,7 @@
 require 'spec_helper_acceptance'
 require_relative './version.rb'
 
-describe 'apache::mod::passenger class', if: fact('osfamily') == 'Debian' do
```
Comment (tphoney) [inline spec/acceptance/mod_passenger_spec.rb]: this needs to go back in,

File: spec/acceptance/mod_php_spec.rb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper_acceptance'
 require_relative './version.rb'
```
Comment (tphoney) [inline spec/acceptance/mod_php_spec.rb]: this needs to go back in

File: spec/acceptance/init_task_spec.rb [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # run a test task
 require 'spec_helper_acceptance'
 
-describe 'apache tasks', if: puppet_version =~ %r{(5\.\d\.\d)} && fact('operatingsystem') != 'SLES' do
+describe 'apache tasks', if: puppet_version =~ %r{(5\.\d\.\d)} && host_inventory['facter']['os']['name'] != 'sles' do
```
Comment (ekohl) [inline spec/acceptance/init_task_spec.rb]: That puppet regex is rather specific. I suspect it'll break on `5.5.10` I think. Probably good to be safe and use `\d+`

File: spec/acceptance/itk_spec.rb [hunk 8]
```diff
@@ -1,25 +1,25 @@
 require 'spec_helper_acceptance'
 
-case fact('osfamily')
-when 'Debian'
+case os[:family]
+when 'debian'
   service_name = 'apache2'
   majrelease = fact('operatingsystemmajrelease')
   variant = if ['6', '7', '10.04', '12.04'].include?(majrelease)
```
Comment (ekohl) [inline spec/acceptance/itk_spec.rb]: None of these are listed in metadata.json

File: spec/acceptance/itk_spec.rb [hunk 9]
```diff
@@ -1,25 +1,25 @@
...
             end
-when 'RedHat'
-  unless fact('operatingsystemmajrelease') == '5'
+when 'redhat'
+  unless os[:release].to_i == 5
     service_name = 'httpd'
-    majrelease = fact('operatingsystemmajrelease')
-    variant = if ['6'].include?(majrelease)
+    majrelease = os[:release].to_i
+    variant = if [6].include?(majrelease)
```
Comment (david22swan) [inline spec/acceptance/itk_spec.rb]: Why is this seperated into two lines?

File: spec/acceptance/mod_fcgid_spec.rb [hunk 10]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper_acceptance'
 
 describe 'apache::mod::fcgid class', if: ((fact('osfamily') == 'RedHat' && fact('operatingsystemmajrelease') != '5') &&
-                                         !(fact('operatingsystem') == 'OracleLinux' && fact('operatingsystemmajrelease') == '7')) do
+    !(fact('operatingsystem') == 'OracleLinux' && fact('operatingsystemmajrelease') == '7')) do
```
Comment (david22swan) [inline spec/acceptance/mod_fcgid_spec.rb]: This should use ServerSpec

File: spec/acceptance/mod_security_spec.rb [hunk 11]
```diff
@@ -2,8 +2,8 @@
 require_relative './version.rb'
 
 describe 'apache::mod::security class', unless: (fact('osfamily') == 'Debian' && (fact('lsbdistcodename') == 'squeeze' ||
-                                                fact('lsbdistcodename') == 'lucid' || fact('lsbdistcodename') == 'precise' ||
-                                                fact('lsbdistcodename') == 'wheezy')) || (fact('operatingsystem') == 'SLES' && fact('operatingsystemrelease') < '11') do
+    fact('lsbdistcodename') == 'lucid' || fact('lsbdistcodename') == 'precise' ||
+    fact('lsbdistcodename') == 'wheezy')) || (fact('operatingsystem') == 'SLES' && fact('operatingsystemrelease') < '11') do
```
Comment (david22swan) [inline spec/acceptance/mod_security_spec.rb]: Can these be ServerSpec?

**Comment by ekohl** [comment]
One problem I have with serverspec is that some "facts" have the same name, but a different meaning. I think osfamily is one of them: on Ubuntu facter will return `Debian` and serverspec has `ubuntu`.

File: spec/acceptance/itk_spec.rb [hunk 12]
```diff
@@ -1,27 +1,20 @@
...
-  variant = if ['6', '7', '10.04', '12.04'].include?(majrelease)
-              :itk_only
-            else
-              :prefork
-            end
-when 'RedHat'
-  unless fact('operatingsystemmajrelease') == '5'
+  variant = :prefork
+when 'redhat'
+  unless os[:release].to_i == 5
```
Comment (ekohl) [inline spec/acceptance/itk_spec.rb]: Why is this here? Shouldn't it be `variant = os[:release].to_i >= 7 ? :prefork : :itk_only`

File: spec/acceptance/itk_spec.rb [hunk 13]
```diff
@@ -1,27 +1,20 @@
 require 'spec_helper_acceptance'
 
-case fact('osfamily')
-when 'Debian'
+case os[:family]
+when 'debian'
```
Comment (ekohl) [inline spec/acceptance/itk_spec.rb]: Here you are changing it I think. `os[:family]` is `ubuntu` on Ubuntu where `fact('osfamily')` is `Debian`.

Comment (lionce) [reply spec/acceptance/itk_spec.rb]: hey @ekohl thank you for your patience, support and help. 
os[:family] is returning ubuntu -> you're right
 host_inventory['facter']['os']['name'] returns Ubuntu
 host_inventory['facter']['os']['family'] returns Debian -> what we need here
It's a bit confusing. Please correct me if i'm wrong

Comment (ekohl) [reply spec/acceptance/itk_spec.rb]: You're right. I'd consider never using `os` and always using `host_inventory['facter']` (which I didn't know existed).

File: spec/acceptance/itk_spec.rb (hunk 12 above)
Comment (lionce) [reply spec/acceptance/itk_spec.rb]: @ekohl yes, you're right. The current implementation is pretty messed up. It can be changed with your solution ;)

**Comment by tphoney** [comment]
@lionce can you run this through the ad-hoc. Looks good otherwise

**Comment by lionce** [comment]
Done. Adde a screenshot with adhoc report
